### PR TITLE
refactor(select): tailwindify

### PR DESCRIPTION
**Pull request by julio8a**
Supports #1500

### CODE REVIEW

File: src/components/calcite-select/calcite-select.scss [hunk 1]
```diff
@@ -1,106 +1,105 @@
...
+    border-color-1 
+    border-l-0 
+    text-color-2 
+    flex 
+    pointer-events-none 
+    absolute 
+    right-0 
+    bottom-0 
+    top-0
+    py-0 px-2;
```
Comment (jrowlingson) [inline src/components/calcite-select/calcite-select.scss:92]: py-0 needed?

File: src/components/calcite-select/calcite-select.scss [hunk 2]
```diff
@@ -1,106 +1,105 @@
...
+    border-solid
+    border-color-1 
+    border-l-0 
+    text-color-2 
+    flex 
+    pointer-events-none 
+    absolute 
+    right-0 
+    bottom-0 
+    top-0
```
Comment (jrowlingson) [inline src/components/calcite-select/calcite-select.scss:91]: `inset-y-0`

File: src/components/calcite-select/calcite-select.scss [hunk 3]
```diff
@@ -1,106 +1,105 @@
...
-  padding: 0 var(--calcite-spacing-half);
-  pointer-events: none;
-  position: absolute;
-  right: 0;
-  bottom: 0;
-  top: 0;
+  @apply items-center 
+    bg-transparent 
+    border
+    border-solid
```
Comment (jrowlingson) [inline src/components/calcite-select/calcite-select.scss:82]: Mentioned elsewhere but we will no longer need to set `border-solid`

Comment (julio8a) [reply src/components/calcite-select/calcite-select.scss:82]: okay, I can come back and remove when that's in.

**Comment by julio8a** [comment]
I need some help with this one. I've followed all the values from Figma and worked with @bstifle on this, but for some reason, the medium select is 35px tall when it should be 32px. 
Small (24px) and larger (48px) worked fine.

**Comment by bstifle** [comment]
@julio8a  are you using text--1h ?

File: src/components/calcite-select/calcite-select.scss [hunk 4]
```diff
@@ -1,106 +1,104 @@
...
+    @apply bg-foreground-2;
   }
 }
 
 :host([dir="rtl"]) .select {
-  padding-left: var(--calcite-select-spacing-end);
   padding-right: var(--calcite-select-spacing);
-  border-right: solid 1px var(--calcite-ui-border-1);
-  border-left: none;
+  border-right: solid 1px;
```
Comment (eriklharper) [inline src/components/calcite-select/calcite-select.scss:65]: ☝️ Does `border-r` work to replace this line?

**Comment by julio8a** [comment]
@bstifle, yes

File: src/components/calcite-select/calcite-select.scss (hunk 4 above)
Comment (julio8a) [reply src/components/calcite-select/calcite-select.scss:65]: I cleaned this up.

**Review by jrowlingson — APPROVED** [review]

File: src/components/calcite-select/calcite-select.scss [hunk 5]
```diff
@@ -1,106 +1,107 @@
...
 }
 
 :host([dir="rtl"]) .select {
-  padding-left: var(--calcite-select-spacing-end);
   padding-right: var(--calcite-select-spacing);
-  border-right: solid 1px var(--calcite-ui-border-1);
-  border-left: none;
+  @apply border-0 
+    border-r
+    border-r-solid
```
Comment (caripizza) [inline src/components/calcite-select/calcite-select.scss:68]: @julio8a `border-r-solid` and `border-l-solid` should not be used anymore - that utility is going away per pr #1788. I'll go ahead and remove both of these selectors in this file over in that pr, but please don't use them any more.

**Pull request merged**